odl/util/utility.py

```diff
@@ -417,6 +417,39 @@ def normalized_index_expression(indices, shape, int_to_slice=False):
     return tuple(indices)
 
 
+def snr(signal, noise, impl):
+    """Compute the signal-to-noise ratio.
+
+    Parameters
+    ----------
+    signal : `array-like`
+        Noiseless data.
+    noise : `array-like`
+        Noise.
+    impl : {'general', 'dB'}
+        Implementation method.
+        'general' means SNR = s_power / n_power,
+        'dB' means SNR = 10 * log10 (s_power / n_power).
+
+    Returns
+    -------
+    The value of signal-to-noise ratio.
```

Comment:
```python
snr : `float`
    value of signal-to-noise ratio. 
    If the power of noise is zero, then the return is 'inf',
    otherwise, the computed value.
```